lib/aws_auditor/aws.rb

```diff
@@ -8,68 +8,9 @@ class AwsConfig < Hash
   end
 
   class AWSSDK
-    FILE_NAMES = %w[.aws.yml .fog]
-
-    def self.configuration(environment)
-      @environment = environment
-      load_config
-      if @config[:mfa_serial_number]
-        creds = get_session(@config).credentials
-      else
-        creds = { access_key_id: access_key_id(@config), secret_access_key: secret_access_key(@config) }
-      end
-      AWS.config(creds)
-    end
-
-    def self.access_key_id(config)
-      config[:access_key_id] || config[:aws_access_key_id]
+    def self.authenticate(environment)
+      aws = Aws::SharedCredentials.new(profile_name: environment)
+      Aws.config.update({region: 'us-east-1', credentials: aws})
     end
-
-
-    def self.secret_access_key(config)
-      config[:secret_access_key] || config[:aws_secret_access_key]
-    end
-
-    def self.region(config)
-      config[:region] || 'us-east-1'
-    end
-
-    def self.load_config
-      return @config if @config
-      @config = AwsConfig[YAML.load_file(config_path)]
-      if @config.has_key? @environment
-        @config = @config[@environment]
-      else
-        raise MissingEnvironment, "Could not find AWS credentials for #{@environment} environment"
-      end
-      @config
-    end
-
-    def self.config_path
-      if filepath = FILE_NAMES.detect {|filename| File.exists?(filename)}
-        File.join(Dir.pwd, filepath)
-      else
-        old_dir = Dir.pwd
-        Dir.chdir('..')
-        if old_dir != Dir.pwd
-          config_path
-        else
-          puts "Could not find #{FILE_NAMES.join(' or ')}"; exit
-        end
-      end
-    end
-
-    def self.get_mfa_token
```

Comment:
http://docs.aws.amazon.com/cli/latest/topic/config-vars.html
http://blogs.aws.amazon.com/security/post/Tx3D6U6WSFGOK2H/A-New-and-Standardized-Way-to-Manage-Credentials-in-the-AWS-SDKs

According to these, it sounds like shared credentials uses `aws_session_number` instead of `mfa_serial_number`. I'm not sure if `aws_session_number` is just a new way of saying the same thing as `mfa_serial_number`... but I'm not exactly sure.